core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -395,7 +395,8 @@ message SQLExecutionUIData {
   optional string error_message = 9;
   map<int64, JobExecutionStatus> jobs = 10;
   repeated int64 stages = 11;
-  map<int64, string> metric_values = 12;
+  bool metric_values_is_null = 12;
```

Comment:
Add this field to make the following case tests pass in `SQLAppStatusListenerWithRocksDBBackendSuite`:

- `basic`
- `SPARK-32615,SPARK-33016: SQLMetrics validation after sparkPlanInfo updated in AQE`

Reply:
Refer to 

https://github.com/apache/spark/blob/23e3c9b7c2f08c5350992934cf660de6d2793982/sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusStore.scala#L98-L104

We should distinguish between `null map` and `empty map`, otherwise `metricValues` will not be calculated or will be repeatedly calculated

https://github.com/apache/spark/blob/23e3c9b7c2f08c5350992934cf660de6d2793982/sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusStore.scala#L65-L82